corehq/apps/userreports/models.py

```diff
@@ -211,6 +219,7 @@ class DataSourceConfiguration(CachedCouchDocumentMixin, Document, AbstractUCRDat
     disable_destructive_rebuild = BooleanProperty(default=False)
     sql_settings = SchemaProperty(SQLSettings)
     validations = SchemaListProperty(Validation)
+    mirrored_engine_ids = SchemaListProperty(MirroredEngineIds)
```

Comment:
I don't follow why we need `server_environment` here. I understand the usage of it in `StaticDataSourceConfiguration` but an instance of a `DataSourceConfiguration` will only ever be on one env.

I think this should be the following:
```python
class DataSourceConfiguration(...):
     mirrored_engine_ids = ListProperty()

class StaticDataSourceConfiguration(...):
    mirrored_engine_ids = SchemaListProperty(MirroredEngineIds)

    @classmethod
    def _get_datasource_config(cls, static_config, domain):
        ....
        # set `mirrored_engine_ids` based on server env
        doc['mirrored_engine_ids'] = get_mirrored_engine_ids(static_config)
        return DataSourceConfiguration.wrap(doc)
```

Reply:
Ha, right. Fixed it.